packages/web-ui/src/providers/CommitProvider.tsx

```diff
@@ -0,0 +1,36 @@
+'use client'
+
+import { createContext, ReactNode, useContext } from 'react'
+
+import { HEAD_COMMIT } from '@latitude-data/core/browser'
+
+interface CommitContextType {
+  commitUuid: string | typeof HEAD_COMMIT
+  isDraft: boolean
+}
+
+const CommitContext = createContext<CommitContextType>({
+  commitUuid: HEAD_COMMIT,
+  isDraft: false,
+})
```

Comment:
`useCurrentCommit` will return both the `commitUuid` and an `isDraft` property, which can be used to allow or forbid modifications depending on the commit.